lib/Controller/PlayerSoftware.php

```diff
@@ -0,0 +1,418 @@
+                    'text' => __('Permissions')
+                );
+            }
+
+            // Download
+            $version->buttons[] = array(
+                'id' => 'content_button_download',
+                'linkType' => '_self', 'external' => true,
+                'url' => $this->urlFor('library.download', ['id' => $media->mediaId]) . '?attachment=' . $media->fileName,
+                'text' => __('Download')
+            );
+        }
+
+        if (count($versions) <= 0)
+            throw new NotFoundException('Version file not found', 'PlayerSoftware');
+
+        $this->getState()->template = 'grid';
+        $this->getState()->recordsTotal = $this->playerVersionFactory->countLast();
+        $this->getState()->setData($versions);
+    }
+
+    /**
+     * Version Delete Form
+     * @param int $versionId
+     * @throws NotFoundException
+     * @throws XiboException
+     */
+    public function deleteForm($versionId)
+    {
+        $version = $this->playerVersionFactory->getById($versionId);
+        $media = $this->mediaFactory->getById($version->mediaId);
+
+        if (!$this->getUser()->checkDeleteable($media))
+            throw new AccessDeniedException();
+
+        $version->load();
+
+        $this->getState()->template = 'playersoftware-form-delete';
+        $this->getState()->setData([
+            'version' => $version,
+            'help' => $this->getHelp()->link('Player Software', 'Delete')
+        ]);
+    }
+
+    /**
+     * Delete Version
+     * @param int $versionId
+     *
+     * @throws NotFoundException
+     * @throws XiboException
+     * @SWG\Delete(
+     *  path="/playersoftware/{versionId}",
+     *  operationId="playerSoftwareDelete",
+     *  tags={"Player Software"},
+     *  summary="Delete Version",
+     *  description="Delete Version file from the Library and Player Versions table",
+     *  @SWG\Parameter(
+     *      name="versionId",
+     *      in="path",
+     *      description="The Version ID to Delete",
+     *      type="integer",
+     *      required=true
+     *   ),
+     *  @SWG\Response(
+     *      response=204,
+     *      description="successful operation"
+     *  )
+     * )
+     */
+    public function delete($versionId)
+    {
+        /** @var PlayerVersion $version */
+        $version = $this->playerVersionFactory->getById($versionId);
+        /** @var Media $media */
+        $media = $this->mediaFactory->getById($version->mediaId);
+
+        if (!$this->getUser()->checkDeleteable($media))
+            throw new AccessDeniedException();
+
+        $version->load();
+        $media->load();
+
+        // Delete
+        $version->delete();
+        $media->setChildObjectDependencies($this->layoutFactory, $this->widgetFactory, $this->displayGroupFactory, $this->displayFactory, $this->scheduleFactory, $this->playerVersionFactory);
+        $media->delete();
```

Comment:
you probably want to delete the media record before you delete the version record, because mediaId is a foreign key on version